cli/src/main/java/org/jboss/as/cli/embedded/EmbedServerHandler.java

```diff
@@ -0,0 +1,316 @@
+                // Poll for server state. Alternative would be to get ControlledProcessStateService
+                // and do reflection stuff to read the state and register for change notifications
+                long expired = bootTimeout == null ? Long.MAX_VALUE : System.nanoTime() + bootTimeout;
+                String status = "starting";
+                final ModelNode getStateOp = new ModelNode();
+                getStateOp.get(ClientConstants.OP).set(ClientConstants.READ_ATTRIBUTE_OPERATION);
+                getStateOp.get(ClientConstants.NAME).set("server-state");
+                do {
+                    try {
+                        final ModelNode response = mcc.execute(getStateOp);
+                        if (Util.isSuccess(response)) {
+                            status = response.get(ClientConstants.RESULT).asString();
+                        }
+                    } catch (Exception e) {
+                        // ignore and try again
+                    }
+
+                    if ("starting".equals(status)) {
+                        try {
+                            Thread.sleep(50);
+                        } catch (InterruptedException e) {
+                            Thread.currentThread().interrupt();
+                            throw new CommandLineException("Interrupted while waiting for embedded server to start");
+                        }
+                    } else {
+                        break;
+                    }
+                } while (System.nanoTime() < expired);
+
+                if ("starting".equals(status)) {
+                    assert bootTimeout != null; // we'll assume the loop didn't run for decades
+                    // Disconnecting will trigger the event listener above, to stop server and restore environment
+                    ctx.disconnectController();
+                    throw new CommandLineException("Embedded server did not exit 'starting' status within " +
+                            TimeUnit.MILLISECONDS.toSeconds(bootTimeout) + " seconds");
+                }
+
+            }
+            ok = true;
+        } catch (ServerStartException e) {
+            throw new CommandLineException("Cannot start embedded server", e);
+        } finally {
+            if (!ok) {
+                ctx.disconnectController();
+                restorer.restoreEnvironment();
+            } else if (contextSelector != null) {
+                contextSelector.restore(null);
+            }
+        }
+    }
+
+    private void configureLogContext(LogContext embeddedLogContext, File jbossHome, CommandContext ctx) {
+        File standaloneDir =  new File(jbossHome, "standalone");
+        File configDir =  new File(standaloneDir, "configuration");
+        File logDir =  new File(standaloneDir, "log");
+        File bootLog = new File(logDir, "boot.log");
+        File loggingProperties = new File(configDir, "logging.properties");
```

Comment:
Makes sense. I didn't notice the --std-out stuff until after I made the comment :)